Source/PINCache.m

```diff
@@ -62,6 +64,7 @@ - (instancetype)initWithName:(NSString *)name
                                              keyDecoder:nil
                                          operationQueue:_operationQueue];
         _memoryCache = [[PINMemoryCache alloc] initWithOperationQueue:_operationQueue];
+        _observerManager = [[PINCacheKeyObserverManager alloc] initWithCache:self];
```

Comment:
I wonder if this could be initialized on demand, that way if the feature wasn't used it wouldn't generate overhead? Below you could check `block || _observerManager` before adding the completion too.